tests/Export/Adapters/PriceAdapterTest.php

```diff
@@ -213,4 +224,223 @@ public function testProductPriceWithCurrency(): void
+                    $netCustomerGroupId => 3,
+                    $grossCustomerGroupId => 15
+                ]
+            ],
+            'Test unit config and no advanced prices' => [
+                'groupsData' => $this->getAdvancedPricesTestGroupData(
+                    [
+                        ['groupId' => $netCustomerGroupId, 'displayGross' => false, 'prices' => []],
+                        ['groupId' => $grossCustomerGroupId, 'displayGross' => true, 'prices' => []]
+                    ]
+                ),
+                'advancedPricingConfig' => 'unit',
+                'expectedPrices' => [
+                    $netCustomerGroupId => 10,
+                    $grossCustomerGroupId => 15
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider advancedPricesTestProvider
+     */
+    public function testIsCorrectAdvancedPriceIsExported(
+        array $groupsData,
+        string $advancedPricingConfig,
+        array $expectedPrices
+    ): void {
+        if (Utils::versionLowerThan('6.4.9.0')) {
+            $this->markTestSkipped('Advanced price calculation by Product entity exists in newer Shopware versions');
+        }
+
+        /** @var ExportContext $exportContext */
+        $exportContext = $this->getContainer()->get('fin_search.export_context');
+
+        $config = $this->getFindologicConfig(['advancedPricing' => $advancedPricingConfig]);
+        $config->initializeBySalesChannel($this->salesChannelContext);
+        $configShopkey = 'ABCDABCDABCDABCDABCDABCDABCDABCD';
+        $this->enableFindologicPlugin($this->getContainer(), $configShopkey, $this->salesChannelContext);
+
+        $adapter = new PriceAdapter(
+            $this->salesChannelContext,
+            $this->exportContext,
+            $this->getContainer()->get(ProductPriceCalculator::class),
+            $this->getContainer()->get(CustomerGroupContextProvider::class),
+            $config
+        );
+
+        $customerGroups = $this->generateCustomers($groupsData);
+        $exportContext->setCustomerGroups($customerGroups);
+        $this->createRules($groupsData);
+        $productEntity = $this->createTestProduct(['prices' => $this->getPrices($groupsData)]);
+        $this->createCustomerGroups($groupsData);
+        $this->createCustomers($groupsData);
+
+        $prices = $adapter->adapt($productEntity);
+
+        foreach ($customerGroups as $customerGroup) {
+            $userGroup = Utils::calculateUserGroupHash($exportContext->getShopkey(), $customerGroup->getId());
+
+            foreach ($prices as $price) {
+                foreach ($price->getValues() as $group => $value) {
+                    if ($userGroup === $group) {
+                        $this->assertEquals($expectedPrices[$customerGroup->getId()], $value);
+                    }
+                }
+            }
+        }
+    }
+
+    private function getAdvancedPricesTestGroupData(array $groupsData): array
```

Comment:
Any specific reason, why this is not part of `AdvancedPriceHelper`?

Reply:
I've added it to AdvancedPriceHelper